pkg/datarepair/checker/online.go

```diff
@@ -0,0 +1,72 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package checker
+
+import (
+	"context"
+	"time"
+
+	"storj.io/storj/pkg/overlay"
+	"storj.io/storj/pkg/pb"
+	"storj.io/storj/pkg/storj"
+)
+
+// ReliableCache caches the reliable nodes for the specified staleness duration
+// and updates automatically from overlay.
+type ReliableCache struct {
+	overlay    *overlay.Cache
+	staleness  time.Duration
+	lastUpdate time.Time
+	reliable   map[storj.NodeID]struct{}
```

Comment:
I would prefer to have a map[storj.NodeID]bool.
Reason for this is simply that if a node got an upload with the 5 minute timeframe and then that file is checked, we would account that node as down (not reliable) right?

Reply:
The check for missing pieces also handles the piece creation date:

```
missingPieces, err := checker.nodestate.MissingPieces(ctx, createdAt, pieces)

[snip]

func (cache *ReliableCache) MissingPieces(ctx context.Context, created time.Time, pieces []*pb.RemotePiece) ([]int32, error) {
	if created.After(cache.lastUpdate) || time.Since(cache.lastUpdate) > cache.staleness {
		err := cache.Refresh(ctx)
		if err != nil {
			return nil, err
		}
	}
```

So if we encounter a piece that is newer than our last update then we'll refresh the cache.

Reply:
Using a bool could work also I guess, but would require updating one node at a time.

Reply:
Oh, true. Good call.